test/livebook_web/plugs/auth_plug_test.exs

```diff
@@ -59,6 +67,41 @@ defmodule LivebookWeb.AuthPlugTest do
       assert conn.status == 200
       assert conn.resp_body =~ "New notebook"
     end
+
+    @tag token: "grumpycat"
+    test "redirects to referer on valid authentication", %{conn: conn} do
+      referer = "/import?url=example.com"
+
+      conn =
+        conn
+        |> merge_private(plug_session: %{})
+        |> put_session(:redirect_to, referer)
```

Comment:
There's [`init_test_session`](https://hexdocs.pm/plug/Plug.Test.html#init_test_session/2) :)

That said, what about a single integration test for the redirect flow, something like this:

```elixir
referer = "/import?url=example.com"

conn = get(conn, referer)
assert redirected_to(conn) == "/authenticate"

conn = post(conn, "/authenticate", password: "grumpycat")
assert redirected_to(conn) == referer
```

Reply:
That's really good 😀